storage/Filesystems/FilesystemImpl.h

```diff
@@ -252,6 +253,32 @@ namespace storage
 
     }
 
+
+    class EnsureMounted : boost::noncopyable
+    {
+
+    public:
+
+	/**
+	 * Ensures that the filesystem is mounted somewhere.
+	 *
+	 * The mode is not enforced.
```

Comment:
If the device is already mounted the mode is ignored since the class does not do a second mount.